internal/pkg/cli/deploy/svc.go

```diff
@@ -279,6 +286,11 @@ func newWorkloadDeployer(in *WorkloadDeployerInput) (*workloadDeployer, error) {
 	}, nil
 }
 
+// AddonsStackBuilder returns the StackBuilder for this workload's addons.
+func (w *workloadDeployer) AddonsStackBuilder() StackBuilder {
```

Comment:
Was there a reason why you preferred returning an interface? 
> “Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values”
> https://www.efekarakus.com/golang/2019/12/29/working-with-interfaces-in-go.html

Can we return `*addon.Addons` and let the client define the `StackBuilder` interface? (see comment below for how to do unit-testing)
```go
func (w *workloadDeployer) Addons() *addon.Addons
```

Reply:
The main reason I went this route was to avoid other packages (`pkg/cli`) needing an explicit dependency on `pkg/addon`, instead being able to only depend on `pkg/cli/deploy`, which they already do.

The problem is that `workloadDeployer` is _already an interface_ in `pkg/cli`, so it'll need a new function like on `Addons() *addon.Addons` on that interface.

I'm fine switching - i also don't love breaking accept-interfaces-return-structs; we'll just have to have an explicit dependency on `pkg/addon` in `pkg/cli` and `pkg/cli/deploy` (which is fine 😊)

Reply:
actually - i think i have a better solution! instead, i'll just change the function to `AddonsTemplate() (string, error)` instead.

Reply:
:shipit: